src/shared.c

```diff
@@ -814,6 +822,50 @@ _Bool GlobMatch(const char *pattern, const char *text) {
 }
 
 /**
+ * @brief Matches the pattern against specified text, returning true if the pattern
+ * matches, false otherwise.
+ *
+ * A match means the entire string TEXT is used up in matching.
+ *
+ * In the pattern string, `*' matches any sequence of characters,
+ * `?' matches any character, [SET] matches any character in the specified set,
+ * [!SET] matches any character not in the specified set.
+ *
+ * A set is composed of characters or ranges; a range looks like
+ * character hyphen character(as in 0-9 or A-Z).
+ * [0-9a-zA-Z_] is the set of characters allowed in C identifiers.
+ * Any other character in the pattern must be matched exactly.
+ *
+ * To suppress the special syntactic significance of any of `[]*?!-\',
+ * and match the character exactly, precede it with a `\'.
+ */
+_Bool GlobMatch(const char *pattern, const char *text) {
+	return GlobMatch_(pattern, text, true);
+}
+
+/**
+ * @brief Matches the pattern against specified text case-insensitively, returning true
+ * if the pattern matches, false otherwise.
+ *
+ * A match means the entire string TEXT is used up in matching.
+ *
+ * In the pattern string, `*' matches any sequence of characters,
+ * `?' matches any character, [SET] matches any character in the specified set,
+ * [!SET] matches any character not in the specified set.
+ *
+ * A set is composed of characters or ranges; a range looks like
+ * character hyphen character(as in 0-9 or A-Z).
+ * [0-9a-zA-Z_] is the set of characters allowed in C identifiers.
+ * Any other character in the pattern must be matched exactly.
+ *
+ * To suppress the special syntactic significance of any of `[]*?!-\',
+ * and match the character exactly, precede it with a `\'.
+ */
+_Bool GlobiMatch(const char *pattern, const char *text) {
+	return GlobMatch_(pattern, text, false);
+}
```

Comment:
ideally we should get rid of glob and use proper regexes, like from objectively or something, but yeah I can do that